kit/feature/feature.go

```diff
@@ -0,0 +1,133 @@
+package feature
+
+import (
+	"context"
+	"strings"
+
+	"github.com/opentracing/opentracing-go"
+)
+
+type contextKey string
+
+const featureContextKey contextKey = "influx/feature/v1"
+
+// Flagger returns flag values.
+type Flagger interface {
+	// Flags returns a map of flag keys to flag values.
+	//
+	// If an authorization is present on the context, it may be used to compute flag
+	// values according to the affiliated user ID and its organization and other mappings.
+	// Otherwise, they should be computed generally or return a default.
+	//
+	// One or more flags may be provided to restrict the results.
+	// Otherwise, all flags should be computed.
+	Flags(context.Context, ...Flag) (map[string]interface{}, error)
+}
+
+// Annotate the context with a map computed of computed flags.
+func Annotate(ctx context.Context, f Flagger, flags ...Flag) (context.Context, error) {
+	computed, err := f.Flags(ctx, flags...)
+	if err != nil {
+		return nil, err
+	}
+
+	span := opentracing.SpanFromContext(ctx)
+	if span != nil {
+		for k, v := range computed {
+			span.LogKV(k, v)
+		}
+	}
+
+	return context.WithValue(ctx, featureContextKey, computed), nil
+}
+
+// FlagsFromContext returns the map of flags attached to the context
+//by Annotate, or nil if none is found.
```

Comment:
nit here and line 55:
```suggestion
// by Annotate, or nil if none is found.
```